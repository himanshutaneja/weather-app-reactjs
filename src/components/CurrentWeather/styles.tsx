import styled from 'styled-components';

export const CurrentWeatherContainer = styled.div`
  background-image: linear-gradient(#046f94, #3493ad);
  padding: 20px;
  border-radius: 6px;
  box-shadow: 0 0 12px 0 rgb(0 0 0 / 20%);
  margin-bottom: 20px;
`;

export const Header = styled.div`
  font-size: 1.25rem;
  font-weight: 400;
`;

export const Location = styled.div`
  font-weight: 600;
  font-size: 1.25rem;
`;

export const Time = styled.div`
  font-size: 1.125rem;
  opacity: 0.65;
`;

export const Detail = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const Temperature = styled.span`
  font-size: 6rem;
  line-height: 0.96em;
  font-weight: 500;
`;

export const Phrase = styled.div`
  font-size: 1.5rem;
  font-weight: 700;
`;

export const WeatherIcon = styled.i`
  font-size: 80px;
`;

export const TemperatureHighLow = styled.div`
  font-size: 1.5rem;
  font-weight: 700;
  text-align: right;
  margin: 12px;
`;
