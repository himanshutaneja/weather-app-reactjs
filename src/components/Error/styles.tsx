import styled from 'styled-components';

export const ErrorContainer = styled.div`
  background-image: linear-gradient(#046f94, #3493ad);
  margin: 0 0 1px;
  margin: 20px;
  padding: 10px;
  border-radius: 5px;
  color: #fff;
`;
