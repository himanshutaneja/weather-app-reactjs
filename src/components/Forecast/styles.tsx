import styled from 'styled-components';

export const ForcastContainer = styled.div`
  display: flex;
  padding: 20px;
  border-radius: 6px;
  box-shadow: 0 0 12px 0 rgb(0 0 0 / 20%);
  background-image: linear-gradient(#046f94, #3493ad);
  justify-content: space-evenly;
`;

export const ForcastItemContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const Day = styled.h3`
  font-size: 1.125rem;
  line-height: 1.66em;
  border-bottom: 1px solid;
`;

export const TempMax = styled.div`
  font-size: 2rem;
`;

export const TempMin = styled.div`
  font-size: 1.125rem;
`;

export const WeatherIcon = styled.i`
  font-size: 2.5rem;
  margin: 10px 0 15px 0;
`;

export const Rain = styled.div`
  font-size: 1.125rem;
`;
