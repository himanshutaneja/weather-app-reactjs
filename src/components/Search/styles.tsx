import styled from 'styled-components';

export const SearchBox = styled.div`
  background-color: #046f94;
`;

export const SearchBarWrapper = styled.div`
  padding: 11px;
  display: flex;
  align-items: center;
  height: 50px;
  margin: 0 400px;
`;

export const SearchBar = styled.input`
  padding: 5px 6px;
  border: 0;
  border-radius: 20px;
  background: hsla(0, 0%, 100%, 0.2);
  text-align: center;
  font-size: 1rem !important;
  color: #fff;
  width: 100%;
  cursor: text;

  ::placeholder {
    color: #fff;
  }
`;
