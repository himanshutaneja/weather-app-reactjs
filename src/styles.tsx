import styled from 'styled-components';

export const Layout = styled.div`
  background-color: #046f94;
  background-image: linear-gradient(#69a9bf, #a3d4dd, #f8fcfd);
  height: 100vh;
`;
