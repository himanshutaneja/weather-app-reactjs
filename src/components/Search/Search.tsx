import React, { FC, KeyboardEvent, ChangeEvent } from 'react';
import { SearchBox, SearchBarWrapper, SearchBar } from './styles';
interface SearchProps {
  placeholderText: string;
  onSearch: (searchParam: string) => void;
}

const Search: FC<SearchProps> = ({ placeholderText, onSearch }) => {
  const [value, setValue] = React.useState('');
  const [placeholder, setPlaceholder] = React.useState(placeholderText);

  const handleChangeValue = (e: ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
  };

  const handleKeyPress = (e: KeyboardEvent) => {
    if (e.key === 'Enter') {
      onSearch(value);
      setValue('');
    }
  };

  return (
    <SearchBox>
      <SearchBarWrapper>
        <SearchBar
          type="text"
          placeholder={placeholder}
          onChange={handleChangeValue}
          value={value}
          onKeyPress={handleKeyPress}
          onFocus={() => setPlaceholder('')}
          onBlur={() => setPlaceholder(placeholderText)}
        />
      </SearchBarWrapper>
    </SearchBox>
  );
};

export default Search;
