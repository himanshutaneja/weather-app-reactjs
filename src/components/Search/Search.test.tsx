import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { act } from 'react-dom/test-utils';
import Search from './Search';
import { SearchBar } from './styles';

describe('Search', () => {
  let wrapper: ShallowWrapper;

  const onSearchMock = jest.fn();
  const setStateMock = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');

  beforeEach(() => {
    jest.clearAllMocks();
    //@ts-ignore
    useStateSpy.mockImplementation(init => [init, setStateMock]);
    wrapper = shallow(
      <Search placeholderText="Enter search term..." onSearch={onSearchMock} />,
    );
  });

  it('should render search bar correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should clear placeholder on focus', () => {
    act(() => {
      wrapper.find(SearchBar).simulate('focus');
    });

    expect(setStateMock).toHaveBeenCalledWith('');
  });

  it('should reset placeholder to input prop on blur', () => {
    act(() => {
      wrapper.find(SearchBar).simulate('blur');
    });

    expect(setStateMock).toHaveBeenCalledWith('Enter search term...');
  });

  it('should change input value of controlled input element', () => {
    act(() => {
      wrapper
        .find(SearchBar)
        .simulate('change', { target: { value: 'new search term' } });
    });

    expect(setStateMock).toHaveBeenCalledWith('new search term');
  });

  it('should invoke onSearch handler on enter key press', () => {
    act(() => {
      wrapper.find(SearchBar).simulate('keypress', { key: 'Enter' });
    });

    expect(onSearchMock).toHaveBeenCalled();
  });
});
