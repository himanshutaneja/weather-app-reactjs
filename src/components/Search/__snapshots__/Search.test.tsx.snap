// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render search bar correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.input
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      placeholder="Enter search term..."
      type="text"
      value=""
    />
  </styled.div>
</styled.div>
`;
